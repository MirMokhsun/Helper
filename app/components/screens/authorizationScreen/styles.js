import { StyleSheet } from 'react-native';
import Utils from '../../../api/utils/utils';

export const styles = StyleSheet.create({
    scrollview: {
        width: "100%",
    },
    wrapperAbsolute: {
        position: "absolute",
        width: "100%",
        height: "100%",
        top: 0,
        left: 0,
        backgroundColor: '#E5E5E5',
    },
    container: {
        width: "90%",
        minHeight: Utils.size.height * 0.5, // 50% need for IphoneX
        flexDirection: 'column',
        alignItems: 'center',
        backgroundColor: '#FFFFFF',
        borderRadius: 10,
        marginVertical: 20,
        justifyContent: 'space-around',
    },
    title: {
        fontFamily: 'Roboto',
        fontWeight: 'bold',
        lineHeight: 32,
        fontSize: 22,
        textAlign: 'center',
        color: '#000756',
        marginVertical: 25,
    },
    backgroundBlue: {
        position: 'absolute',
        top: -50,
        left: 0
    },
    logoWrapper: {
        alignItems: 'center',
        justifyContent: 'center',
        height: 35,
        width: '100%',
        marginVertical: 20,
    },
    buttonGoBack: {
        width: 25,
        height: 25,
        marginLeft: 20,
        alignItems: 'center',
        justifyContent: 'center',
    },
    buttonGoBackWrapper: {
        width: '100%',
        height: 25,
        justifyContent: 'flex-start',
        marginBottom: 10,
        marginTop: Utils.isIphoneX ? 50 : 20,
    }
})