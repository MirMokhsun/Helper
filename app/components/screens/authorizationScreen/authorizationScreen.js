import React, { Component } from 'react';
import { View, ScrollView, Text, KeyboardAvoidingView, StatusBar } from 'react-native';
import { styles } from './styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { CommonInput } from '../../commonComponents/commonInput/commonInput';
import { filterEmail, filterPassword } from '../../../api/regexFilters/regexFilters';
import { localization } from '../../../common/localization';
import ButtonSubmit from '../../commonComponents/buttonSubmit/buttonSubmit';
import Background from '../../commonComponents/svg/Background';
import Logo from '../../commonComponents/svg/logo';
import ButtonForgotPassword from './authoriztionElements/buttonForgotPassword/buttonForgotPassword';
import { setAuthorizationEmail, setAuthorizationPassword } from './reducer/authorizationDataAction';
import InputPasswordWithSVG from './authoriztionElements/inputPasswordWithSVG/inputPasswordWithSVG';
import { showAuthorizationSpiner } from '../../commonComponents/buttonSubmit/redux/spinersAction';
import ButtonBack from '../../commonComponents/buttonBack/buttonBack';
import { crmlogin } from '../../../api/autorization/autorizationSingletone';
import Utils from '../../../api/utils/utils';

const { isIOS } = Utils;

export class AuthorizationScreen extends Component {
    static navigationOptions = () => {
        return {
            header: null,
        }
    };

    constructor(props) {
        super(props);

        this.state = {
            isAuthorizationButtomDisabled: true,
        }
    }

    UNSAFE_componentWillReceiveProps(nextProps) {
        const { email, password } = nextProps.authorizationData;
        if (email.isValid && password.isValid ) {
            this.setState({ isAuthorizationButtomDisabled: false });
        } else {
            this.setState({ isAuthorizationButtomDisabled: true });
        }
    }

    UNSAFE_componentWillUnmount() {
        const { setAuthorizationEmail, setAuthorizationPassword, } = this.props;
        setAuthorizationEmail({ isValid: false, value: null });
        setAuthorizationPassword({ isValid: false, value: null });
    }

    sendAuthorizationData = () => {
        const { showAuthorizationSpiner, authorizationData: { email, password }, navigation } = this.props;
        showAuthorizationSpiner();
        crmlogin.login(email.value, password.value, navigation);
    }

    goBack = () => {
        const { navigation } = this.props;
        navigation.goBack();
    }

    render() {
        const { authorizationData: { password, email }, setAuthorizationEmail, setAuthorizationPassword, authorizationSpiner, navigation, language } = this.props;
        const { isAuthorizationButtomDisabled } = this.state;
        const local = localization[language];
        return (
            <KeyboardAvoidingView style={styles.wrapperAbsolute} behavior={isIOS ? "padding" : null} enabled >
                <StatusBar barStyle="light-content" backgroundColor="#000756"/>
                <ScrollView style={styles.scrollview} keyboardShouldPersistTaps="handled" contentContainerStyle={{ alignItems: 'center', }}>
                    <View style={styles.backgroundBlue}>
                        <Background />
                    </View>
                    <View style={styles.buttonGoBackWrapper}>
                        <ButtonBack id="AuthorizationID" navigation={navigation} />
                    </View>
                    <View style={styles.logoWrapper}>
                        <Logo />
                    </View>
                    <View style={styles.container} >
                        <Text style={styles.title}>{local.LogInToTheAccount}</Text>
                        <CommonInput
                            keyboardType="email-address"
                            placeholder={local.placeHolder.email}
                            errorText={local.errors.errorEmail}
                            filter={filterEmail}
                            action={setAuthorizationEmail}
                            isValid={email.isValid}
                            testID="InputEmailAutorizationID"
                            textErrorID="InputErrorEmailAutorizationID"
                            currentLanguageName={language}
                        />
                        <InputPasswordWithSVG
                            placeholder={local.placeHolder.password}
                            errorText={local["Password failed"]}
                            filter={filterPassword}
                            action={setAuthorizationPassword}
                            isValid={password.isValid}
                            testID="InputPasswordWithSVGID"
                            textErrorID="InputPasswordWithSVGErrorID"
                            currentLanguageName={language}
                        />
                        <ButtonForgotPassword text={local["Forgot password?"]} />
                        <ButtonSubmit
                            text={local.enter}
                            testID="ButtonSubmitAutorizationID"
                            isDisabled={isAuthorizationButtomDisabled}
                            isShowSpiner={authorizationSpiner}
                            onPress={this.sendAuthorizationData}
                        />
                    </View>
                </ScrollView>
            </KeyboardAvoidingView>
        )
    }
}

const mapStateToProps = state => ({
    language: state.currentLanguage.language,
    authorizationData: state.authorizationData,
    authorizationSpiner: state.spiners.authorizationSpiner,
    isConnected: state.connectionStatus.isConnected,
});

const mapDispatchToProps = dispatch => bindActionCreators({
    setAuthorizationEmail,
    setAuthorizationPassword,
    showAuthorizationSpiner,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AuthorizationScreen);