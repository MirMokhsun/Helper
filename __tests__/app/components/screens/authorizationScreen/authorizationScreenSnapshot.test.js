import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { AuthorizationScreen } from '../../../../../app/components/screens/authorizationScreen/authorizationScreen';

it('create snapshot AuthorizationScreen', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<AuthorizationScreen
        setAuthorizationEmail={() => { }}
        setAuthorizationPassword={() => { }}
        authorizationSpiner = { true }
        authorizationData = { {password:'password', email:"email" } }
        forgotPasswordData= { {email:"email"} }
        language ={"ru"}
    />);
    expect(toolbarSnapshot).toMatchSnapshot();
});