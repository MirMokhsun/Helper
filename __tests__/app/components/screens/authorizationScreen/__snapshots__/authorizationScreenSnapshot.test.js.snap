// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot AuthorizationScreen 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "#E5E5E5",
      "height": "100%",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <StatusBar
    animated={false}
    backgroundColor="#000756"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <ScrollViewMock
    contentContainerStyle={
      Object {
        "alignItems": "center",
      }
    }
    keyboardShouldPersistTaps="handled"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "top": -50,
        }
      }
    >
      <Background />
    </View>
    <View
      style={
        Object {
          "height": 25,
          "justifyContent": "flex-start",
          "marginBottom": 10,
          "marginTop": 50,
          "width": "100%",
        }
      }
    >
      <withNavigation(ButtonBack)
        id="AuthorizationID"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 35,
          "justifyContent": "center",
          "marginVertical": 20,
          "width": "100%",
        }
      }
    >
      <Logo />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "marginVertical": 20,
          "minHeight": 667,
          "width": "90%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000756",
            "fontFamily": "Roboto",
            "fontSize": 22,
            "fontWeight": "bold",
            "lineHeight": 32,
            "marginVertical": 25,
            "textAlign": "center",
          }
        }
      >
        Войти в аккаунт
      </Text>
      <CommonInput
        action={[Function]}
        currentLanguageName="ru"
        errorText="Ошибка ввода e-mail"
        filter={/\\^\\(\\(\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\(\\\\\\.\\[\\^<>\\(\\)\\\\\\[\\\\\\]\\\\\\\\\\.,;:\\\\s@"\\]\\+\\)\\*\\)\\|\\("\\.\\+"\\)\\)@\\(\\(\\\\\\[\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\.\\[0-9\\]\\{1,3\\}\\\\\\]\\)\\|\\(\\(\\[a-zA-Z\\\\-0-9\\]\\+\\\\\\.\\)\\+\\[a-zA-Z\\]\\{2,\\}\\)\\)\\$/}
        keyboardType="email-address"
        placeholder="Email"
        testID="InputEmailAutorizationID"
        textErrorID="InputErrorEmailAutorizationID"
      />
      <InputPasswordWithSVG
        action={[Function]}
        currentLanguageName="ru"
        errorText="Ошибка ввода пароля"
        filter={/\\.\\{6,\\}/}
        placeholder="Пароль"
        testID="InputPasswordWithSVGID"
        textErrorID="InputPasswordWithSVGErrorID"
      />
      <withNavigation(ButtonForgotPassword)
        text="Забыли пароль?"
      />
      <ButtonSubmit
        isDisabled={true}
        isShowSpiner={true}
        onPress={[Function]}
        testID="ButtonSubmitAutorizationID"
        text="Вход"
      />
    </View>
  </ScrollViewMock>
</KeyboardAvoidingView>
`;
